chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -837,7 +848,39 @@ public synchronized void loadChunks(TaskTracker taskTracker, World world, Collec
 
         loadedChunks.add(cp);
 
-        world.getChunk(cp).getChunkData(chunkData, palette);
+        try {
+          nextChunkDataTask.get();
+        } catch(InterruptedException ignored) { // If except, load the chunk synchronously
+          if(usingFirstChunkData) {
+            world.getChunk(chunkPositions[i]).getChunkData(chunkData1, palette);
+          }
+          else {
+            world.getChunk(chunkPositions[i]).getChunkData(chunkData2, palette);
+          }
+        } catch(ExecutionException e) {
+          throw new RuntimeException(e.getCause());
+        }
+
+        ChunkData chunkData; //the chunk data to be used for THIS iteration
+        {
+          ChunkData nextChunkData; //the chunk data to be used for the next iteration
+          if (usingFirstChunkData) {
+            chunkData = chunkData1;
+            nextChunkData = chunkData2;
+          } else {
+            chunkData = chunkData2;
+            nextChunkData = chunkData1;
+          }
+          usingFirstChunkData = !usingFirstChunkData;
+
+          if (i + 1 < chunkPositions.length) { //if has next request next
+            int finalI = i;
```

Comment:
```suggestion
            final int finalI = i;
```

Just to be sure/make the intent more clear? :thinking: